lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java

```diff
@@ -112,6 +113,7 @@
     private static final String SET_SYSTEM_SESSION_PROPERTY_ACCESS_DENIED_MESSAGE = "Access Denied: Cannot set system session property .*";
     private static final String SET_CATALOG_SESSION_PROPERTY_ACCESS_DENIED_MESSAGE = "Access Denied: Cannot set catalog session property .*";
 
+    @SuppressWarnings("deprecation")
```

Comment:
What and why do we suppress? Maybe we can fix it by not using deprecated API?

Reply:
There is a `checkCanSetUser` method in `SystemAccessControl`, which is deprecated. The replacement is supposed to be `checkCanImpersonateUser`but it's not a direct replacement and I'm not yet sure what's the difference in their use cases.

Reply:
Yeah... this let's not suppress this. We need to remember about `checkCanSetUser` and eventually remove that method.

Let's remove this suppression.

Reply:
The proper fix is here: https://github.com/trinodb/trino/pull/4081

Reply:
OK, cool, I'll leave it then. Thanks!